NBitcoin/Utils.cs

```diff
@@ -699,10 +698,39 @@ public static IPEndPoint ParseIpEndpoint(string endpoint, int defaultPort)
 			}
 			catch(FormatException)
 			{
+#if !WINDOWS_UWP
 				address = Dns.GetHostEntry(ip).AddressList[0];
+#else
+				string adr = DnsLookup(ip).Result;
+				// if not resolved behave like GetHostEntry
+				if (adr == string.Empty)
+					throw new SocketException(11001);
+				else
+					address = IPAddress.Parse(adr);
+#endif
 			}
 			return new IPEndPoint(address, port);
 		}
+
+#if WINDOWS_UWP
+		private static async Task<string> DnsLookup(string remoteHostName)
+		{
+			IReadOnlyList<EndpointPair> data = await DatagramSocket.GetEndpointPairsAsync(new HostName(remoteHostName), "0");

```

Comment:
here you have to add .ConfigureAwait(false); or you'll endup with deadlock in context like ASP.NET or WPF, and likely UWP if there is a user interface.